module/VuFind/src/VuFind/Controller/HoldsTrait.php

```diff
@@ -282,4 +283,57 @@ public function holdAction()
         $view->setTemplate('record/hold');
         return $view;
     }
+
+    /**
+     * Email the successful request to staff as a notification.
+     *
+     * @param array $holdDetails Details about the successful hold request
+     *
+     * @return void
+     */
+    protected function emailRequestPlaced($holdDetails)
+    {
+        $config = $this->getConfig();
+        $to = $this->getEmailRecipient($holdDetails);
+        $from = $config->Catalog->holds_email_from ?? null;
+        if (!$to || !$from) {
+            return;
+        }
+
+        $renderer = $this->getViewRenderer();
+        $message = $renderer->render(
+            'Email/request-placed.phtml',
+            ['hold_details' => $holdDetails]
+        );
+
+        $subject = $this->translate('request_email_subject', [
+            '%%id%%' => $holdDetails['id'] ?? '',
+            '%%barcode%%' => $holdDetails['barcode'] ?? '',
+        ]);
+        try {
+            $mailer = $this->getService(\VuFind\Mailer\Mailer::class);
+            $mailer->send(
+                $to,
+                $from,
+                $subject,
+                $message
+            );
+        } catch (\VuFind\Exception\Mail $e) {
+            return;
+        }
+    }
+
+    /**
+     * Get the appropriate staff email recipient for a hold request.  This falls back
+     * on a configured default but may be overridden to vary based on the hold details.
+     *
+     * @param array $holdDetails Details about the successful hold request
+     *
+     * @return string The email address
+     */
+    protected function getEmailRecipient($holdDetails)
```

Comment:
The logic I'll be implementing locally is to look at the pickup location and do a mapping from that to a particular email address.  Not sure if that is generic enough to share.

Reply:
Could that be generalized to something like a prioritized list of $holdDetails properties mapped to email addresses? This is pretty ugly, but something like:

```
[EmailNotificationMap]
property_name[value] = "email@example.com"
other_property_name[value2] = "email2@example.com"
default = "sharedbox@example.com"
```

(Obviously this would lend itself better to YAML or JSON configuration, but if we have to use .ini, it might work).